### PR TITLE
Clarify to support Preempt-RT kernel

### PR DESCRIPTION
**Pull request by TiejunChina**
Signed-off-by: Tiejun Chen <tiejun.china@gmail.com>

------
v4:
* Fix one missing change in .mailmap
v3:
* Keep .mailmap sorted alphabetically
v2:
* Separate the last commit
* Update AUTHOR
v1:
* As promised in #2897, I submit this script as well. 
* Add one example yml file, and mention this support in docs/kernels.md and will maintain this part in the future.

### CODE REVIEW

File: MAINTAINERS [hunk 1]
```diff
@@ -164,6 +164,7 @@ on disputes for technical matters."
 			"justincormack",
 			"riyazdf",
 			"rn",
+			"TiejunChina",
```
Comment (rn) [inline MAINTAINERS]: This is the MAINTAINERS file, not the the `.mailmap` file as I suggested you to modify.

Can this also be a separate commit, not lumped in with unrelated changes?

File: scripts/prefix-with-number.sh [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+# This can help get patches prefixed by the term of number, like 0001-xxxx.patch, 0002-yyyy.patch, etc.
```
Comment (rn) [inline scripts/prefix-with-number.sh]: can you give a sample invocation for this script, ie how to actually use it?

File: MAINTAINERS (hunk 1 above)
Comment (thaJeztah) [reply MAINTAINERS]: If your preferred e-mail address is `tiejun.china@gmail.com`, you can add this line to the `.mailmap`:

```
Tiejun Chen <tiejun.china@gmail.com> <tiejunc@vmware.com>
```

Patch:

```patch
diff --git a/.mailmap b/.mailmap
index 588c67bc6..1a666bd07 100644
--- a/.mailmap
+++ b/.mailmap
@@ -49,5 +49,6 @@ Sebastiaan van Stijn <sebastiaan.vanstijn@docker.com> <github@gone.nl>
 Simon Ferquel <simon.ferquel@docker.com> <simon.ferquel@hotmail.fr>
 Thomas Gazagnaire <thomas.gazagnaire@docker.com> <thomas@gazagnaire.com>
 Thomas Gazagnaire <thomas.gazagnaire@docker.com> <thomas@gazagnaire.org>
+Tiejun Chen <tiejun.china@gmail.com> <tiejunc@vmware.com>
 Vincent Demeester <Vincent.Demeester@docker.com> <vincent@sbr.pm>
 Vincent Demeester <Vincent.Demeester@docker.com> <vdemeester@docker.com>
```

That will updated the existing entry in the `AUTHORS` file when it's re-generated. You can re-generate that file using `scripts/generate-authors.sh`

File: scripts/prefix-with-number.sh (hunk 2 above)
Comment (TiejunChina) [reply scripts/prefix-with-number.sh]: @rn Sure.

File: MAINTAINERS (hunk 1 above)
Comment (TiejunChina) [reply MAINTAINERS]: @rn @thaJeztah Okay.

File: .mailmap [hunk 3]
```diff
@@ -51,3 +51,4 @@ Thomas Gazagnaire <thomas.gazagnaire@docker.com> <thomas@gazagnaire.com>
 Thomas Gazagnaire <thomas.gazagnaire@docker.com> <thomas@gazagnaire.org>
 Vincent Demeester <Vincent.Demeester@docker.com> <vincent@sbr.pm>
 Vincent Demeester <Vincent.Demeester@docker.com> <vdemeester@docker.com>
+Tiejun Chen <tiejun.china@gmail.com> <tiejunc@vmware.com>
```
Comment (thaJeztah) [inline .mailmap]: Could you keep this list sorted alphabetically?

Comment (TiejunChina) [reply .mailmap]: @thaJeztah Got it.

File: examples/rt-for-vmware.yml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  - name: getty
+    image: linuxkit/getty:v0.2
+    env:
+     - INSECURE=true
+  - name: rngd
+    image: linuxkit/rngd:v0.2
+  - name: dhcpcd
+    image: linuxkit/dhcpcd:v0.2
+  - name: open-vm-tools
+    image: linuxkit/open-vm-tools:30b93dde3bbcc2346f01a570b932b768b04fc54c
```
Comment (rn) [inline examples/rt-for-vmware.yml]: Is there a reason you are not using the `v0.2` tag here?

File: examples/rt-for-vmware.yml [hunk 5]
```diff
@@ -0,0 +1,36 @@
+kernel:
+  image: linuxkit/kernel:4.14.18-rt
+  cmdline: "console=tty0"
+init:
+  - linuxkit/init:v0.2
+  - linuxkit/runc:v0.2
+  - linuxkit/containerd:v0.2
```
[the remark this answers is not in the document]
Comment (TiejunChina) [reply examples/rt-for-vmware.yml]: @rn Okay. Sorry, I didn't realize these items were alreayd changed.

File: examples/rt-for-vmware.yml (hunk 4 above)
Comment (TiejunChina) [reply examples/rt-for-vmware.yml]: @rn Will set that as v0.2.

File: .mailmap (hunk 3 above)
Comment (rn) [reply .mailmap]: This request is still not addressed. Please keep the list sorted

**Comment by TiejunChina** [comment]
@rn Sorry, I missed this part :(
